docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,237 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Charmed Kubeflow Databases Migration Guide\n",
+    "\n",
+    " When upgrading to Charmed Kubeflow 1.8 these databases must be migrated to MySQL. There are two databases in Charmed Kubeflow - `katib-db` and `kfp-db`. In Charmed Kubeflow 1.7 these databases are backed by `charmed-osm-mariadb-k8s` charm. In the same release (1.7) we introduced support for MySQL databases, which can be deployed using the `mysql-k8s` charm. In Charmed Kubeflow 1.8 release MariaDB is not supported. Databases must be migrated prior to upgrading to Charmed Kubeflow 1.8."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "- Client machine with access to deployed Charmed Kubeflow 1.7.\n",
+    "- Juju version is 2.9.\n",
+    "- Verify that migration of databases is required by checking types of databases deployed. If database charm deployed is `mariadb-k8s` migration is required:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "    # replace DB_CHARM with desired unit name: `katib-db` or `kfp-db`\n",
+    "    DB_CHARM= kfp-db | katib-db\n",
+    "    juju show-application kfp-db | yq '.[] | .charm'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "- Enough storage in the cluster to support backup/restore of the databases.\n",
+    "- Enough storage to store database data for backup on client machine.\n",
+    "- `mysql-client` on client machine (install by running `sudo apt install mysql-client`)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Obtain existing database credentials\n",
+    "\n",
+    "To obtain database credential execute the following commands for each database to be migrated and use those credentials in migration steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# replace DB_UNIT with desired unit name: `katib-db` or `kfp-db`\n",
+    "DB_UNIT= katib-db | kfp-db\n",
+    "\n",
+    "# obtain username and password of existing MariadDB database from DB relation\n",
+    "# in Charmed Kubeflow 1.6 and 1.7 username used is `root` and password is specified in `mysql` relation by\n",
```

Comment:
I wouldn't mention CKF 1.6 as we'll stop supporting that version pretty soon.

Reply:
+1